File: src/bakir-store.rs
use std::process::Command;
use std::env;

fn main() {
    let args: Vec<String> = env::args().collect();
    if args.len() < 2 || args[1] == "-h" || args[1] == "--help" {
        display_help();
        return;
    }
    if args.len() < 3 { return; }
    let action = &args[1];
    let package = &args[2];
    match action.as_str() {
        "-i" | "--install" => install_logic(package),
        _ => display_help(),
    }
}

fn display_help() {
    println!("━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━");
    println!("   🏪 متجر باكير الذكي | Bakir Store v4.7   ");
    println!("━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━");
    println!("🛡️  [الأدوات السيادية المتاحة]:");
    
    // جلب القائمة الديناميكية من السحاب
    let output = Command::new("curl")
        .args(&["-s", "https://raw.githubusercontent.com/abuhussen/Bakir-Core/main/remote-repo/apps.json"])
        .output();
    
    if let Ok(out) = output {
        let s = String::from_utf8_lossy(&out.stdout);
        for line in s.lines() {
            if line.contains("\"name\"") {
                let name = line.split('"').nth(3).unwrap_or("");
                println!("   • {}", name);
            }
        }
    }

    // إضافة الأدوات التي نثبت وجودها يدوياً لضمان الظهور الدائم
    println!("   • bakir-opt             (أداة التحسين ⚡)");
    println!("   • bakir-terminal-theme  (هوية النظام 🎨)");

    println!("\n🌐  [الأدوات العالمية]:");
    println!("   • vlc          • timeshift    • stacer");
    println!("   • firefox      • vscode       • gimp");
    
    println!("\n💡 للتثبيت: bakir-store -i [الاسم]");
    println!("━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━");
}

fn install_logic(pkg: &str) {
    println!("🔍 فحص المستودعات لـ [{}]...", pkg);
    let url = format!("https://raw.githubusercontent.com/abuhussen/Bakir-Core/main/remote-repo/{}", pkg);
    let check = Command::new("curl").args(&["-s", "--head", "--fail", &url]).status();

    if check.is_ok() && check.unwrap().success() {
        install_sovereign(pkg, &url);
    } else {
        install_global(pkg);
    }
}

fn install_sovereign(pkg: &str, url: &str) {
    let dest = format!("/usr/bin/{}", pkg);
    println!("🚀 جلب أداة سيادية: [{}]...", pkg);
    let status = Command::new("sudo").args(&["wget", "-q", "--show-progress", url, "-O", &dest]).status();
    if status.is_ok() && status.unwrap().success() {
        let _ = Command::new("sudo").args(&["chmod", "+x", &dest]).status();
        println!("✅ تم التثبيت بنجاح.");
    }
}

fn install_global(pkg: &str) {
    println!("🌐 جلب أداة عالمية: [{}]...", pkg);
    let _ = Command::new("sudo").args(&["apt", "install", "-y", pkg]).status();
}

File: src/bakir-store.go
package main

import (
	"encoding/json"
	"fmt"
	"io/ioutil"
	"os"
)

type App struct {
	Name string `json:"name"`
	Desc string `json:"desc"`
}

type Repo struct {
	BakirCore   []App `json:"bakir_core"`
	GlobalTools []App `json:"global_tools"`
}

func main() {
	fmt.Println("--------------------------------------------------")
	fmt.Println("📦 Bakir Store | متجر باكير السيادي v3.1")
	fmt.Println("--------------------------------------------------")

	// قراءة قاعدة البيانات
	data, err := ioutil.ReadFile("/home/bakir/Bakir-Project/remote-repo/apps.json")
	if err != nil {
		fmt.Println("❌ خطأ: تعذر الوصول لقاعدة بيانات المتجر.")
		return
	}

	var repo Repo
	json.Unmarshal(data, &repo)

	// القسم الأول: الأدوات الأساسية
	fmt.Println("\n📂 [القسم الأول - Bakir Core Tools]")
	for i, app := range repo.BakirCore {
		fmt.Printf("%d. %-20s | %s\n", i+1, app.Name, app.Desc)
	}

	// القسم الثاني: الأدوات العالمية
	fmt.Println("\n🌍 [القسم الثاني - Global Tools]")
	for i, app := range repo.GlobalTools {
		fmt.Printf("%d. %-20s | %s\n", i+1, app.Name, app.Desc)
	}

	fmt.Println("--------------------------------------------------")
}
